engines/director/lingo/lingo-builtins.cpp

```diff
@@ -1774,30 +1774,54 @@ void LB::b_window(int nargs) {
 
 void LB::b_numberofchars(int nargs) {
 	Datum d = g_lingo->pop();
+	d.toString();
 
-	warning("STUB: b_numberofchars");
-	d.toInt();
-	d.u.i = 0;
+	int len = strlen(d.u.s->c_str());
+	delete d.u.s;
 
+	d.u.i = len;
+	d.type = INT;
 	g_lingo->push(d);
 }
 
 void LB::b_numberofitems(int nargs) {
 	Datum d = g_lingo->pop();
 
-	warning("STUB: b_numberofitems");
-	d.toInt();
-	d.u.i = 0;
+	d.toString();
+	int numberofitems = 1;
+	Common::String contents = d.u.s->c_str();
+	uint32 i = 0;
+	while (i < d.u.s->size()){
```

Comment:
Why not `for` here?

Reply:
No specific reason for this code. 
Mainly that other code in Scummvm that dealt with strings were using while loops.
I've converted both `while`s to `for`s.